google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -87,24 +110,81 @@ public boolean isSupportAlreadyAdded(@NotNull Module module,
   /** Initializes the Flexible facet by settings the default paths for app.yaml and Dockerfile
    * and generating the necessary run configurations.
    */
-  public void setupFacet(@NotNull AppEngineFlexibleFacet facet,
-      @NotNull ModifiableRootModel rootModel) {
+  public static void addSupport(@NotNull AppEngineFlexibleFacet facet,
+      @NotNull ModifiableRootModel rootModel,
+      boolean generateConfigFiles) {
     // Allows suggesting app.yaml and Dockerfile locations in facet and deployment UIs.
     VirtualFile[] contentRoots = rootModel.getContentRoots();
     AppEngineProjectService appEngineProjectService = AppEngineProjectService.getInstance();
     if (contentRoots.length > 0) {
-      facet.getConfiguration().setAppYamlPath(
+      Path appYamlPath = Paths.get(
           appEngineProjectService.getDefaultAppYamlPath(contentRoots[0].getPath()));
-      facet.getConfiguration().setDockerfilePath(
+      Path dockerfilePath = Paths.get(
           appEngineProjectService.getDefaultDockerfilePath(contentRoots[0].getPath()));
+
+      facet.getConfiguration().setAppYamlPath(appYamlPath.toString());
+      facet.getConfiguration().setDockerfilePath(dockerfilePath.toString());
+
+      // Configuration file generation.
+      Project project = facet.getModule().getProject();
+      Optional<Path> defaultAppYaml =
+          new CloudSdkAppEngineHelper(project)
+              .defaultAppYaml(FlexibleRuntime.java);
+
+      if (generateConfigFiles) {
+        if (Files.exists(appYamlPath)) {
+          int override = Messages.showYesNoDialog(project,
+              GctBundle.message("appengine.support.appyaml.existing"),
+              GctBundle.message("appengine.support.appyaml.existing.title"),
+              GoogleCloudToolsIcons.APP_ENGINE
+          );
+
+          if (override == Messages.YES) {
+            defaultAppYaml.ifPresent(
+                appYaml -> WriteCommandAction.runWriteCommandAction(project,
+                    () -> {
+                      VirtualFile appYamlVirtualFile = contentRoots[0]
+                          .findFileByRelativePath("/src/main/appengine/app.yaml");
+                      if (appYamlVirtualFile != null) {
+                        PsiFile appYamlPsiFile =
+                            PsiManager.getInstance(project).findFile(appYamlVirtualFile);
+                        if (appYamlPsiFile != null) {
+                          Document appYamlDocument =
+                              PsiDocumentManager.getInstance(project)
+                                  .getDocument(appYamlPsiFile);
+                          if (appYamlDocument != null) {
+                            try {
+                              appYamlDocument.setText(
```

Comment:
I think the problem with FileUtil is that it performs the kind of low level file-system IO operation that we don't want. For example, it makes use of Files/Paths, not Documents/PsiFiles. I think going down that route would lead to similar issues with warnings, and isn't the best practice, but correct me if I'm wrong.